src/main/ruby/truffleruby/core/proc.rb

```diff
@@ -127,4 +129,12 @@ def <<(other)
       end
     end
   end
+
+  private
+
+  def check_if_callable!(value)
+    return if value.kind_of?(Proc) || value.kind_of?(Method) || value.respond_to?(:call)
```

Comment:
Both Proc and Method `respond_to?(:call)` and `respond_to?()` inline caches in TruffleRuby, so there is no need to check explicit types here.

Given it's so simple after that, I would skip the helper and have the logic inline `raise unless ...`

Reply:
Hm, I thought that anyway `kind_of?` is a bit faster than `respond_to?` even if the method is inlined. Or it is not so significant?

Reply:
I would think both compile down to a single check on the metaclass.
You can verify with IGV or https://github.com/Shopify/seafoam if you like, or just write a benchmark using benchmark-ips :)

Reply:
```ruby
require 'benchmark/ips'

VALUE = -> {}

Benchmark.ips do |x|
  x.report 'kind_of?' do
    VALUE.kind_of?(Proc)
  end
  
  x.report 'respond_to?' do
    VALUE.respond_to?(:call)
  end
end
```

```
ruby bench_kind_of_respond_to.rb
Warming up --------------------------------------
            kind_of?   415.314M i/100ms
         respond_to?   434.209M i/100ms
Calculating -------------------------------------
            kind_of?      4.246B (± 1.7%) i/s -     21.596B in   5.088181s
         respond_to?      4.224B (± 0.4%) i/s -     21.276B in   5.036533s
```
So they both seem to optimize away.

Also more checks would be bad if the `value` is actually not a Proc or Method, and different `value` classes have been seen, then those would just be extra needless checks on the fast path.

Reply:
Thanks to the benchmarks! I have no more arguments - I pushed the version with `respond_to?` only 

> Given it's so simple after that, I would skip the helper and have the logic inline raise unless ...

I did it; however, in MRI we have 5 places with this check and maybe we'll return this later; I guess additional method is not affect anything, just readability